### PR TITLE
ISPN-13079 Lock contention in DefaultCacheManager.getCache()

### PR DESCRIPTION
**Pull request by danberindei**
https://issues.redhat.com/browse/ISPN-13079

### CODE REVIEW

File: core/src/main/java/org/infinispan/manager/DefaultCacheManager.java [hunk 1]
```diff
@@ -722,6 +722,9 @@ public void start() {
     * @param block {@code true} when we need all the global components to be running.
     */
    private void internalStart(boolean block) {
+      if (status == ComponentStatus.RUNNING)
```
Comment (wburns) [inline core/src/main/java/org/infinispan/manager/DefaultCacheManager.java:725]: Should this be 
```java
if (status != ComponentStatus.INSTANTIATED) {
   return;
}
```
like the code inside the lock?

Comment (wburns) [reply core/src/main/java/org/infinispan/manager/DefaultCacheManager.java:725]: Nevermind, forgot we have to wait for INITIALIZING to complete :)

**Comment by wburns** [comment]
Integrated into master, thanks @danberindei !

**Comment by tristantarrant** [comment]
I guess we need a backport

**Comment by reda-alaoui** [comment]
Hello,

Can this be backported please?

**Comment by tristantarrant** [comment]
https://github.com/infinispan/infinispan/pull/9366

**Pull request merged**